code/modules/mob/dead/observer/observer.dm

```diff
@@ -300,6 +300,20 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 				A.overlays += source
 				source.layer = old_layer
 				source.plane = old_plane
+		spawn(0)
```

Comment:
Instead of using `spawn` I recommending creating a proc out of the code you've added here, something like `proc/reenter_corpse_prompt()` (you can name it whatever you want).

And then call the proc with `INVOKE_ASYNC(src, .proc/reenter_corpse_prompt)`